pypeit/coadd2d.py

```diff
@@ -1164,44 +1207,72 @@ def get_brightest_obj(self, specobjs_list, spat_ids):
                 - snr_bar: ndarray, float, shape (len(list),): Average
                   S/N over all the orders for this object
         """
+        msgs.info('Finding brightest object')
         nexp = len(specobjs_list)
         nspec = specobjs_list[0][0].TRACE_SPAT.shape[0]
         nslits = spat_ids.size
 
-        slit_snr_max = np.full((nslits, nexp), -np.inf)
+        slit_snr_max = np.full((nslits, nexp), np.nan)
```

Comment:
Rather than assigning SNR to nan and using nanmean and nanargmax below,  it is better coding practice to create an explicit good object mask for keeping track of good objects in this routine. The point is that nans could end up passed bag due to other bugs in the compute_weights routine. Your current code would not distinguish between that possible bug and objects that are missed because they were on bad slits. Whereas if you create an explicit mask, you would distinguish between those cases and identify the bug that was returning nans elsewhere. In general implicit masking using nans is dangerous and it is always preferable to use explicit masking since you can catch other things that return nans when they break.

Reply:
I agree with this.

Reply:
Done. Thank you!